test/stream_test.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:test/test.dart';
+import 'dart:async';
+import 'entity.dart';
+import 'test_env.dart';
+import 'objectbox.g.dart';
+import 'package:objectbox/src/observable.dart';
+
+// ignore_for_file: non_constant_identifier_names
+
+void main() {
+  TestEnv env;
+  Box box;
+
+  setUp(() {
+    env = TestEnv('streams');
+    box = env.box;
+  });
+
+  test('Subscribe to stream of entities', () async {
+
+    final result = <String>[];
+    final text = TestEntity_.tString;
+    final condition = text.notNull();
+    final query = box.query(condition).order(text).build();
+    final queryStream = query.findStream();
+    final subscription = queryStream.listen((list) {
+      final str = list.map((t) => t.tString).toList().join(', ');
+      result.add(str);
+    });
+
+    box.put(TestEntity(tString: 'Hello world'));
+    await Future.delayed(Duration(seconds: 0)); // ffi explodes without
```

Comment:
The docs for Future.delayed contain

> If the duration is 0 or less, it completes no sooner than in the next event-loop iteration, after all microtasks have run.

I don't know much about Dart, but this sounds like the observer callback is held up from executing while this "thread" is active. Might this cause issues in an actual app where the callback execution is held back too long? Is it possible to run them on a separate "thread" so they are not held up by other code?

Reply:
Hm, reading the `StreamController.broadcast` docs on using `sync` is false mode this actually sounds like expected behavior (events may be delivered at any point after the code adding the event has completed). LGTM then.

Reply:
Maybe remove that `ffi explodes without` comment than and clarify regarding delayed event delivery.

Reply:
I updated this.